pandas/tests/frame/methods/test_nlargest.py

```diff
@@ -209,3 +209,13 @@ def test_nlargest_multiindex_column_lookup(self):
         result = df.nlargest(3, ("x", "b"))
         expected = df.iloc[[3, 2, 1]]
         tm.assert_frame_equal(result, expected)
+
+    def test_nlargest_nan_column(self):
+        # GH#43060
+        df = pd.DataFrame(
+            {"grp": [1, 1, 2, 2], "y": [1, 0, 2, 5], "z": [1, 2, np.nan, np.nan]}
+        )
+        dfgrp = df.groupby("grp")
```

Comment:
this looks like it is a test for groupby/apply, can you make a test specific to DataFrame.foo

Reply:
If you mean as a general test, there are already plenty of test cases in the same file.

Reply:
i mean tests in this file generally shouldn't depend on groupby logic.  can you write this test using DataFrame.nlargest directly without doing a groupby